### PR TITLE
EZ-639: Quick Links Menu block on Moderation Dashboard

### PR DESCRIPTION
**Pull request by saurabh-2k17**

### CODE REVIEW

File: ezcontent_publish.module [hunk 1]
```diff
@@ -60,3 +60,15 @@ function ezcontent_publish_entity_insert(EntityInterface $entity) {
...
   }
 }
+
+/**
+ * Implements hook_theme().
+ * {@inheritdoc} 
+ */
+function ezcontent_publish_theme($existing, $type, $theme, $path) {
+  return [
+    'usershortcutblock_template' => [
```
Comment (arshadkhan53) [inline ezcontent_publish.module]: we should make template name more readable` user_shortcut_block_template`

File: ezcontent_publish.module [hunk 2]
```diff
@@ -60,3 +60,15 @@ function ezcontent_publish_entity_insert(EntityInterface $entity) {
       break;
   }
 }
+
+/**
+ * Implements hook_theme().
+ * {@inheritdoc}
```
Comment (arshadkhan53) [inline ezcontent_publish.module]: Please check if inherit doc is applicable here

File: ezcontent_publish.routing.yml [hunk 3]
```diff
@@ -9,3 +9,19 @@ ezcontent_publish.user_profile:
...
+ezcontent_publish.shortcut.link_add:
+  path: '/usershortcut/{shortcut_set}/add-link'
+  defaults:
+    _controller: '\Drupal\shortcut\Controller\ShortcutController::addForm'
+    _title: 'Add link'
+  requirements:
+    _role: 'authenticated'
+
+ezcontent_publish.shortcut_set.customize_form:
+  path: '/usershortcut/{shortcut_set}/customize'
```
Comment (arshadkhan53) [inline ezcontent_publish.routing.yml]: we can be more specific for path,  like `/usershortcut/{shortcut_set}/list-links`

File: ezcontent_publish.routing.yml [hunk 4]
```diff
@@ -9,3 +9,19 @@ ezcontent_publish.user_profile:
         type: 'entity:user'
   requirements:
     _permission: 'ezcontent publish view user profile'
+
+ezcontent_publish.shortcut.link_add:
+  path: '/usershortcut/{shortcut_set}/add-link'
```
Comment (vaibhavjain-in) [inline ezcontent_publish.routing.yml]: Suggest a better path.

File: src/Plugin/Block/EzContentUserShortcutBlock.php [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    // Getting current user role.
+    $currentUserRoles = $this->account->getRoles();
+    $shortcutObj = [];
+
+    if (!in_array('administrator', $currentUserRoles)) {
+      $currentUserId = $this->account->id();
+      $currentUsername = $this->account->getUsername();
+      $currentShortcutset = $this->entityTypeManager
+        ->getStorage('shortcut_set')
+        ->loadByProperties(['id' => 'shortcut_set_' . $currentUsername . $currentUserId]);
```
Comment (vaibhavjain-in) [inline src/Plugin/Block/EzContentUserShortcutBlock.php]: Can this be
```suggestion
        ->loadByProperties(['id' => 'shortcut_set_' . $currentUsername . '_' . $currentUserId]);
```

File: src/Plugin/Block/EzContentUserShortcutBlock.php [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+      // Getting current user shortcut set.
+      $currentuserShortcutsetId = $this->entityTypeManager
+        ->getListBuilder('shortcut_set');
+
+      if (!$currentShortcutset) {
+        // Creating default shortcut set for new user.
+        $set = $currentuserShortcutsetId->getStorage()->create([
+          'id' => 'shortcut_set_' . $currentUsername . $currentUserId,
+          'label' => 'Default Shortcut Set for ' . $currentUsername,
```
Comment (vaibhavjain-in) [inline src/Plugin/Block/EzContentUserShortcutBlock.php]: t() missing

File: src/Plugin/Block/EzContentUserShortcutBlock.php [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        ->loadByProperties(['id' => 'shortcut_set_' . $currentUsername . $currentUserId]);
+
+      // Getting current user shortcut set.
+      $currentuserShortcutsetId = $this->entityTypeManager
+        ->getListBuilder('shortcut_set');
+
+      if (!$currentShortcutset) {
+        // Creating default shortcut set for new user.
+        $set = $currentuserShortcutsetId->getStorage()->create([
+          'id' => 'shortcut_set_' . $currentUsername . $currentUserId,
```
Comment (vaibhavjain-in) [inline src/Plugin/Block/EzContentUserShortcutBlock.php]: generate and save the ID value in $id, before If condition

File: src/Plugin/Block/EzContentUserShortcutBlock.php [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+      // Loading current user shortcuts for display.
+      $currentUserShortcuts = $this->entityTypeManager
+        ->getStorage('shortcut')
+        ->loadByProperties(['shortcut_set' => 'shortcut_set_' . $currentUsername . $currentUserId]);
+
+      // Building Add link URL.
+      $addShortcutUrl = Url::fromRoute('ezcontent_publish.shortcut.link_add', ['shortcut_set' => 'shortcut_set_' . $currentUsername . $currentUserId]);
+      $addShortcutLink = Link::fromTextAndUrl(t('Add Shortcut'), $addShortcutUrl);
+      $addShortcutLink = $addShortcutLink->toRenderable();
```
Comment (vaibhavjain-in) [inline src/Plugin/Block/EzContentUserShortcutBlock.php]: can we club these 2 lines ?

File: src/Plugin/Block/EzContentUserShortcutBlock.php [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+      // Building Operation link URL.
+      $operationShortcutUrl = Url::fromRoute('ezcontent_publish.shortcut_set.customize_form', ['shortcut_set' => 'shortcut_set_' . $currentUsername . $currentUserId]);
+      $operationShortcutLink = Link::fromTextAndUrl(t('Visit Shortcuts'), $operationShortcutUrl);
+      $operationShortcutLink = $operationShortcutLink->toRenderable();
```
Comment (vaibhavjain-in) [inline src/Plugin/Block/EzContentUserShortcutBlock.php]: Can we club these 2 lines.

File: templates/usershortcutblock-template.html.twig [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<section class="user-shortcuts-block-wrapper">
```
Comment (vaibhavjain-in) [inline templates/usershortcutblock-template.html.twig]: Please rectify formatting issues.

**Review by vaibhavjain-in — CHANGES_REQUESTED** [review]
@saurabh-2k17  Please see changes to be made.

File: src/Plugin/Block/EzContentUserShortcutBlock.php [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    // Getting current user role.
+    $currentUserRoles = $this->account->getRoles();
+    if (!in_array('administrator', $currentUserRoles)) {
+      $currentUserId = $this->account->id();
+      $currentUsername = $this->account->getUsername();
+      $currentShortcutset = $this->entityTypeManager
+      ->getStorage('shortcut_set')
+      ->loadByProperties(['id' => 'shortcut_set_' . $currentUsername . '_' . $currentUserId]);
+      // Getting current user shortcut set.
+      $currentuserShortcutsetId = $this->entityTypeManager
```
Comment (arshadkhan53) [inline src/Plugin/Block/EzContentUserShortcutBlock.php]:
```suggestion
      $entityTypeManager= $this->entityTypeManager
```

**Review by arshadkhan53 — CHANGES_REQUESTED** [review]

**Review by arshadkhan53 — APPROVED** [review]
LGTM

**Pull request merged**